types.ts

```diff
@@ -463,6 +470,46 @@ export interface MultiQueryParamTokenAuthentication extends BaseAuthentication {
   }>;
 }
 
+/**
```

Comment:
This doc says only tokenUrl may be necessary. I saw we had a bunch of other parameters under the OAuth2Authentication that may be relevant to token exchange so included them here in a shared interface